### PR TITLE
Introduce exist assertion

### PR DESCRIPTION
**Pull request by idegtiarenko**
It is common to check if a field exists in the response json (regardless the value) in yaml tests. Today this is done using `is_true` assertion if the value is not "0" otherwise assertion is failing and need to be replaced with either `gte` or `is_false`. This change introduces the `exist` assertion that allows to verify the field exists regardless its value.

### CODE REVIEW

File: test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/ExistAssertion.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Represents an exists assert section:
+ *
+ *   - exists:  get.fields.bar
```
Comment (DaveCTurner) [inline test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/ExistAssertion.java:24]: The actual assertion is `exist:`. I'd rather it was `exists:` tho.

File: test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/ExistAssertion.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ */
+public class ExistAssertion extends Assertion {
+
+    private static final Logger logger = LogManager.getLogger(IsTrueAssertion.class);
```
Comment (DaveCTurner) [inline test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/ExistAssertion.java:29]:
```suggestion
    private static final Logger logger = LogManager.getLogger(ExistAssertion.class);
```

File: test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/ExistAssertion.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    protected void doAssert(Object actualValue, Object expectedValue) {
+        logger.trace("assert that field [{}] exists with any value", getField());
+        String errorMessage = errorMessage();
+        assertThat(errorMessage, actualValue, notNullValue());
+    }
+
+    private String errorMessage() {
+        return "field [" + getField() + "] does not exists";
```
Comment (DaveCTurner) [inline test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/ExistAssertion.java:47]: wording nit

```suggestion
        return "field [" + getField() + "] does not exist";
```

**Review by DaveCTurner — COMMENTED** [review]
Nice idea. I left a couple of nits. Also could you add some docs to `rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc` please?

**Comment by idegtiarenko** [comment]
What team label should I put on this pr? Is it Core/Infra?

**Comment by DaveCTurner** [comment]
`:Delivery/Build` I reckon

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.desired_balance/10_basic.yml [hunk 4]
```diff
@@ -73,38 +73,38 @@ setup:
...
-  - is_true: 'cluster_balance_stats.tiers.data_content.actual_disk_usage.min'
-  - is_true: 'cluster_balance_stats.tiers.data_content.actual_disk_usage.max'
-  - is_true: 'cluster_balance_stats.tiers.data_content.actual_disk_usage.average'
-  - is_true: 'cluster_balance_stats.tiers.data_content.actual_disk_usage.std_dev'
-  - is_true: 'cluster_balance_stats.nodes'
-  - is_true: 'cluster_balance_stats.nodes.$node_name'
-  - gte: { 'cluster_balance_stats.nodes.$node_name.shard_count' : 0 }
-  - gte: { 'cluster_balance_stats.nodes.$node_name.forecast_write_load': 0.0 }
-  - gte: { 'cluster_balance_stats.nodes.$node_name.forecast_disk_usage_bytes' : 0 }
-  - gte: { 'cluster_balance_stats.nodes.$node_name.actual_disk_usage_bytes' : 0 }
```
Comment (DaveCTurner) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.desired_balance/10_basic.yml:107]: I think we want to leave these as `gte` assertions - that way we're saying that these things are present _and_ numeric _and_ nonnegative.

File: test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/AssertionTests.java [hunk 5]
```diff
@@ -170,4 +171,18 @@ public void testInvalidCloseTo() throws Exception {
         exception = expectThrows(IllegalArgumentException.class, () -> CloseToAssertion.parse(parser));
         assertThat(exception.getMessage(), equalTo("value is missing or not a number"));
     }
+
+    public void testExists() throws IOException {
+        parser = createParser(YamlXContent.yamlXContent, "get.fields._timestamp");
+
+        ExistsAssertion trueAssertion = ExistsAssertion.parse(parser);
```
Comment (DaveCTurner) [inline test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/AssertionTests.java:178]: copy/paste nit:

```suggestion
        ExistsAssertion existsAssertion = ExistsAssertion.parse(parser);
```

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.desired_balance/10_basic.yml [hunk 6]
```diff
@@ -172,7 +172,7 @@ setup:
   - do:
       _internal.get_desired_balance: { }
 
-  - gte: { stats.computed_shard_movements: 0 }
+  - exists: 'stats.computed_shard_movements'
```
Comment (DaveCTurner) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.desired_balance/10_basic.yml:175]: Likewise this one should remain `gte` I think.

**Review by DaveCTurner — COMMENTED** [review]
I left a couple more nits. Don't forget those docs in `rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc` either.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**